### ISSUE TITLE
What is the deifnition of "Solid-compliant identity provider"?

### ISSUE THREAD

**Issue by fgro** [solid/node-solid-server#392]
In the docs you write:

`A WebID profile and browser certificate from one of the Solid-compliant identity providers, such as databox.me.`

How do you actually define "Solid-compliant identify" precisely?

Thank you!

**Comment by dmitrizagidulin** [comment]
I think it might help to know the context of your question, to answer this. What are you trying to do?

(I think that line of the README might be somewhat obsolete. You can test out node-solid-server by launching it and having it create a local account.)

**Comment by dmitrizagidulin** [comment]
I think initially, I put that sentence there to point out that - most of the interesting things you can do with Solid (as opposed to a plain web server like Apache) require you to authenticate, which means you need a Solid [WebID profile](https://github.com/solid/solid-spec#webid-profile-documents) on a server that supports some way of authenticating you that is comprehensible to other solid servers.
Which, at the time meant - you need to have created a Solid account which generated a private key and stored it in your browser, and stored the public key in your profile -- the whole WebID+TLS bit.

Does that make more sense?

**Comment by dmitrizagidulin** [comment]
@fgro Does this answer your question?

**Comment by fgro** [comment]
@dmitrizagidulin Thank you very much for your reply. 

The context of my question was that from reading the sentence I got the impression that there was a 3rd party dependency whereas I'm trying to achieve decentralization. But if I understand you correctly all the sentence really means is that the "solid webid profiles" spec must be fulfilled (https://github.com/solid/solid-spec/blob/master/solid-webid-profiles.md) - I could potentially be my own "identity provider" .

That is exactly what I'm currently trying to do/test with my local development environment. Unfortunately "api/accounts/{new|create}" takes me to an external url , has an external dependency. https://github.com/solid/node-solid-server/blob/master/lib/identity-provider.js#L634

I'm still fairly new to all this so maybe I'm misunderstand something.

**Comment by dmitrizagidulin** [comment]
@fgro Ah, makes perfect sense :)

In general, we're trying to do everything we can to achieve complete decentralization, no worries.
And yes, you understood correctly -- the sentence just means filling the solid web profiles spec, and we definitely aim for everybody to be their own identity providers.

As for the external url on account creation - yeah, ignore that part for the moment. We're actually right in the process of replacing that hardcoded link (and have solid-server read the link from your config).
The general idea there is -- the 'create new account' api endpoint is supposed to redirect you to whatever 'create account' app (html + javascript) that you prefer. We're planning to pre-install these core apps (signup, signin, profile manager, etc) right as you're installing solid-server itself. Until we have that feature implemented, we simply redirect to a static app hosted on Github Pages.

So, it's merely a convenience move, not really a third party dependency. And we're working to fix it.
Let me know if you have any other questions.

**Comment by fgro** [comment]
@dmitrizagidulin thanks for your support. One more question that I'd have:

Regarding 2.1.2 on https://www.w3.org/2005/Incubator/webid/spec/tls/ 

... if we wanted to use a self-hosted certficate maker: does the solid server include a "certificate maker"?

**Comment by fgro** [comment]
just a side comment:  section 4 of the document https://www.w3.org/2005/Incubator/webid/spec/tls/#verifying-the-webids is really hard to follow ... while I understand the need for formalization and precision, expressing things overly complicated will not help for a "technology" to find a lot of adoption in the world.

**Comment by dmitrizagidulin** [comment]
> if we wanted to use a self-hosted certficate maker: does the solid server include a "certificate maker"?

@fgro if I understand that section correctly, then yes, it does. solid-server returns the created certificate, and the browser stores it in its keychain.

**Comment by dmitrizagidulin** [comment]
> just a side comment:

Fwiw, we're working on an alternative authentication mechanism (OpenID Connect, specifically). :)

**Comment by mediaprophet** [comment]
Related thoughts have been tabled here: https://lists.w3.org/Archives/Public/public-webid//2016Jun/0003.html
